mne_bids_pipeline/_import_data.py

```diff
@@ -549,106 +548,169 @@ def _get_bids_path_in(
+        task=task,
         kind=kind,
     )
-    in_files = dict()
-    key = f"raw_run-{run}"
-    in_files[key] = bids_path_in
-    _update_for_splits(in_files, key, single=True)
-    if add_bads:
-        _add_bads_file(
+    return _path_dict(
+        cfg=cfg,
+        bids_path_in=bids_path_in,
+        key=key,
+        add_bads=add_bads,
+        kind=kind,
+        allow_missing=allow_missing,
+    )
+
+
+def _get_rest_path(
+    *,
+    cfg: SimpleNamespace,
+    subject: str,
+    session: Optional[str],
+    kind: Literal["orig", "sss"],
+    add_bads: Optional[bool] = None,
+) -> dict:
+    if not (cfg.process_rest and not cfg.task_is_rest):
+        return dict()
+    return _get_run_path(
+        cfg=cfg,
+        subject=subject,
+        session=session,
+        run=None,
+        task="rest",
+        kind=kind,
+        add_bads=add_bads,
+        allow_missing=True,
+    )
+
+
+def _get_noise_path(
+    *,
+    cfg: SimpleNamespace,
+    subject: str,
+    session: Optional[str],
+    kind: Literal["orig", "sss"],
+    mf_reference_run: Optional[str],
+    add_bads: Optional[bool] = None,
+) -> dict:
+    if not (cfg.process_empty_room and get_datatype(config=cfg) == "meg"):
+        return dict()
+    if kind == "sss":
+        raw_fname = _get_bids_path_in(
             cfg=cfg,
-            in_files=in_files,
-            key=key,
+            subject=subject,
+            session=session,
+            run=None,
+            task="noise",
+            kind=kind,
         )
-    if run == cfg.runs[0]:
-        do = dict(
-            rest=cfg.process_rest and not cfg.task_is_rest,
-            noise=cfg.process_empty_room and cfg.datatype == "meg",
+    else:
+        # This must match the logic of _02_find_empty_room.py
+        raw_fname = _get_bids_path_in(
+            cfg=cfg,
+            subject=subject,
+            session=session,
+            run=mf_reference_run,
+            task=get_task(config=cfg),
+            kind=kind,
         )
-        for task in ("rest", "noise"):
-            if not do[task]:
-                continue
-            key = f"raw_{task}"
-            if kind == "sss":
-                raw_fname = bids_path_in.copy().update(run=None, task=task)
-            else:
-                if task == "rest":
-                    raw_fname = bids_path_in.copy().update(run=None, task=task)
-                else:
-                    # This must match the logic of _02_find_empty_room.py
-                    update_kwargs = dict()
-                    if cfg.use_maxwell_filter:
-                        update_kwargs["run"] = cfg.mf_reference_run
-                    raw_fname = _read_json(
-                        _empty_room_match_path(
-                            bids_path_in.copy().update(**update_kwargs),
-                            cfg,
-                        )
-                    )
-                    raw_fname = raw_fname["fname"]
-                    if raw_fname is not None:
```

Comment:
Got rid of 5 (!) levels of indentation by changing/refactoring the logic here. Hopefully it's (much) easier to follow when editing the files now...